components/pixels/lib.rs

```diff
@@ -157,22 +175,73 @@ pub fn load_from_memory(buffer: &[u8], cors_status: CorsStatus) -> Option<Image>
             debug!("{}", msg);
             None
         },
-        Ok(_) => match image::load_from_memory(buffer) {
-            Ok(image) => {
-                let mut rgba = image.into_rgba8();
-                rgba8_byte_swap_colors_inplace(&mut rgba);
-                Some(Image {
-                    width: rgba.width(),
-                    height: rgba.height(),
-                    format: PixelFormat::BGRA8,
-                    bytes: IpcSharedMemory::from_bytes(&rgba),
-                    id: None,
-                    cors_status,
-                })
+        Ok(format) => match format {
+            ImageFormat::Gif => {
+                let decoder = GifDecoder::new(Cursor::new(buffer));
+                let mut width = 0;
+                let mut height = 0;
+                match decoder {
+                    Ok(dec) => {
+                        let mut frames = vec![];
+                        let frames_result = dec.into_frames();
+                        for frame_result in frames_result {
+                            match frame_result {
+                                Ok(mut f) => {
+                                    rgba8_byte_swap_colors_inplace(f.buffer_mut());
+                                    frames.push(ImageFrame {
+                                        bytes: IpcSharedMemory::from_bytes(f.buffer()),
+                                        delay: Some(Duration::from(f.delay())),
+                                    });
+                                    width = f.buffer().width();
+                                    height = f.buffer().height();
```

Comment:
If each frame can be a different size then the `Image` data structure should support that.